components/ai_chat/core/common/yt_util.h

```diff
@@ -26,13 +28,20 @@ inline constexpr auto kYouTubeHosts =
 
 // Extract a caption url from an array of YT caption tracks, from the YT page
 // API.
+COMPONENT_EXPORT(AI_CHAT_COMMON)
```

Comment:
don't export things that should not be public. Move this into a source_set to be accessible in unit tests

Reply:
It's not only used by its own unit tests but also used by `//brave/components/ai_chat/content/browser/page_content_fetcher.cc` and `//brave/components/ai_chat/renderer/page_content_extractor.cc` .
So I moved it to non-public API with visibility limitation in https://github.com/brave/brave-core/pull/29619/commits/2c49b0b2c75eec5aaabc561ee387e7560fd265ea